storage/store.go

```diff
@@ -1584,6 +1591,35 @@ func (s *Store) GroupLocker() sync.Locker {
 	return &s.raftGroupLocker
 }
 
+// CanApplySnapshot implements the multiraft.Storage interface.
+func (s *Store) CanApplySnapshot(rangeID roachpb.RangeID, snap raftpb.Snapshot) bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	if r, ok := s.replicas[rangeID]; ok && r.isInitialized() {
+		// We have the range and it's initialized, so let the snapshot
+		// through. We'll decide later if it's newer or older than our
+		// current state.
+		return true
+	}
+
+	// We don't have the range (or we have an uninitialized
+	// placeholder). Will we be able to create/initialize it?
+	// TODO(bdarnell): can we avoid parsing this twice?
```

Comment:
we can always shell out a "meta" portion of the snapshot to avoid pulling up the whole blob.